crates/driver/src/solver/dto/auction.rs

```diff
@@ -1,81 +1,234 @@
+                    },
+                    reward: order.reward,
+                })
+                .collect(),
+            // TODO Implement this when you do liquidity
+            liquidity: vec![],
+            effective_gas_price: auction.gas_price.into(),
+            deadline: deadline.into(),
+        }
     }
 }
 
+#[serde_as]
 #[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
 pub struct Auction {
+    id: Option<String>,
     tokens: HashMap<H160, Token>,
-    orders: HashMap<usize, Order>,
-    amms: BTreeMap<H160, Amm>,
-    metadata: Option<Metadata>,
+    orders: Vec<Order>,
+    liquidity: Vec<Liquidity>,
+    #[serde_as(as = "serialize::U256")]
+    effective_gas_price: U256,
+    deadline: chrono::DateTime<chrono::Utc>,
 }
 
 #[serde_as]
 #[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
 struct Order {
-    #[serde(skip_serializing_if = "Option::is_none")]
-    id: Option<String>,
+    #[serde_as(as = "serialize::Hex")]
+    uid: [u8; 56],
     sell_token: H160,
     buy_token: H160,
     #[serde_as(as = "serialize::U256")]
     sell_amount: U256,
     #[serde_as(as = "serialize::U256")]
     buy_amount: U256,
-    allow_partial_fill: bool,
-    is_sell_order: bool,
-    fee: TokenAmount,
-    cost: TokenAmount,
-    is_liquidity_order: bool,
-    is_mature: bool,
-    mandatory: bool,
-    has_atomic_execution: bool,
+    #[serde_as(as = "serialize::U256")]
+    fee_amount: U256,
+    kind: Kind,
+    partially_fillable: bool,
+    class: Class,
     reward: f64,
 }
 
-#[serde_as]
 #[derive(Debug, Serialize)]
-struct TokenAmount {
-    #[serde_as(as = "serialize::U256")]
-    amount: U256,
-    token: H160,
+#[serde(rename_all = "lowercase")]
+enum Kind {
+    Sell,
+    Buy,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "lowercase")]
+enum Class {
+    Market,
+    Limit,
+    Liquidity,
 }
 
 #[serde_as]
 #[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
 struct Token {
     decimals: Option<u8>,
-    alias: Option<String>,
-    external_price: Option<f64>,
-    normalize_priority: Option<u64>,
+    symbol: Option<String>,
     #[serde_as(as = "Option<serialize::U256>")]
-    internal_buffer: Option<U256>,
-    accepted_for_internalization: bool,
+    reference_price: Option<U256>,
+    #[serde_as(as = "serialize::U256")]
+    available_balance: U256,
+    trusted: bool,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "kind", rename_all = "lowercase")]
+enum Liquidity {
+    ConstantProduct(ConstantProductPool),
+    WeightedProduct(WeightedProductPool),
+    Stable(StablePool),
+    ConcentratedLiquidity(ConcentratedLiquidityPool),
+    LimitOrder(ForeignLimitOrder),
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ConstantProductPool {
+    id: String,
+    address: H160,
+    #[serde_as(as = "serialize::U256")]
+    gas_estimate: U256,
+    tokens: HashMap<H160, ConstantProductReserve>,
+    fee: f64,
```

Comment:
I think `fee: BigDecimal` or something similar as the OpenAPI spec specifies this as a `Decimal` type, which is an arbitrary precision decimal number encoded as a string (so `"13.37"` and not `13.37`).